extensions/Worker.Extensions.CosmosDB/src/CosmosDBConverter.cs

```diff
@@ -12,12 +12,19 @@
 using Microsoft.Azure.Functions.Worker.Extensions.CosmosDB;
 using Microsoft.Extensions.Options;
 using Microsoft.Extensions.Logging;
+using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
+using System.IO;
 
 namespace Microsoft.Azure.Functions.Worker
 {
     /// <summary>
     /// Converter to bind Cosmos DB type parameters.
     /// </summary>
+    [SupportsDeferredBinding]
+    [SupportsJsonDeserialization]
```

Comment:
`Poco` and `Object` are different. We will be using this definition of Poco for our check -https://learn.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-7.0#pocos-without-parameterless-constructors. And yes, this attribute will cover Poco and Poco collections.

Reply:
Will there be any distinction between arrays and IEnumerable types? I.e. IReadOnlyList<POCO> vs POCO[] Or does this not matter to the converter?

What about for things like the blob trigger where we support POCO but not POCO collections?

```
        [Function(nameof(BlobBookFunction))]
        public void BlobBookFunction(
            [BlobTrigger("book-trigger")] Book data)
        {
            _logger.LogInformation("Id: {id} - Name: {name}", data.Id, data.Name);
        }
```

Trying to think about the impact this will have on the analyzer - like how we can determine that a given collection is supported if not directly listed by `SupportedConverterType` 

E.g. some one tries to bind to `Book[]` on a BlobTrigger - it's not supported. 

Maybe for things like this we just have to build a different analyzer that checks trigger attributes for collection use? Independent of the Supported Types analyzer that will only check `SupportedConverterType` attributes?

Reply:
Discussed offline to remove attribute SupportsJsonDeserialization